Framework/Core/test/test_StaggeringWorkflow.cxx

```diff
@@ -0,0 +1,131 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#include "Framework/WorkflowSpec.h"
+#include "Framework/DataProcessorSpec.h"
+#include "Framework/DataSpecUtils.h"
+#include "Framework/DataAllocator.h"
+#include "Framework/InputRecord.h"
+#include "Framework/InputSpec.h"
+#include "Framework/OutputSpec.h"
+#include "Framework/ControlService.h"
+#include "Framework/OutputRoute.h"
+#include "Framework/Logger.h"
+#include "Framework/SenderPolicy.h"
+#include "Framework/DeviceSpec.h"
+#include <chrono>
+#include <cstring>
+#include <iostream>
+
+void customize(std::vector<o2::framework::SenderPolicy>& policies)
+{
+  // we customize all devices to send data immediately
+  policies.push_back({ "prompt-for-all", [](auto const&) { return true; }, o2::framework::SenderPolicy::SenderOp::WhenReady });
+}
+
+void customize(std::vector<o2::framework::CompletionPolicy>& policies)
+{
+  // we customize the processors to consume data as it comes
+  policies.push_back({ "processor-consume",
+                       [](o2::framework::DeviceSpec const& spec) {
+                         // search for spec names starting with "processor"
+                         return spec.name.find("processor") == 0;
+                       },
+                       [](auto const&) { return o2::framework::CompletionPolicy::CompletionOp::Consume; } });
+}
+
+#include "Framework/runDataProcessing.h"
+
+using namespace o2::framework;
+
+#define ASSERT_ERROR(condition)                                   \
```

Comment:
We should move ASSERT_ERROR to `Framework/Logger`. I think we have it already in a few places.

Reply:
true, will check